### PR TITLE
Base component only allows overlays field

### PR DESCRIPTION
**Pull request by shamsher31**
To fix https://github.com/istio/istio/issues/27240

Based on https://github.com/istio/istio/issues/27240#issuecomment-692059502,  https://github.com/istio/istio/issues/27240#issuecomment-692634227 and [this](https://github.com/istio/istio/blob/4e9e07c3fb322d947e896f88987fa6bf1db0d7d1/operator/cmd/mesh/testdata/manifest-generate/input/pilot_override_kubernetes.yaml#L13).

I'm not sure but I run `protolock commit --force` before I running `make gen`.

### CODE REVIEW

File: operator/v1alpha1/operator.proto [hunk 1]
```diff
@@ -173,8 +173,14 @@ message BaseComponentSpec {
     // Selects whether this component is installed.
     TypeBoolValueForPB enabled = 1;
 
-    // Kubernetes resource spec.
-    KubernetesResourcesSpec k8s = 50;
+    // Overlays for k8s resources.
```
Comment (costinm) [inline operator/v1alpha1/operator.proto:176]: Is it backward compatible ? Upgrade story ?

**Review by costinm — CHANGES_REQUESTED** [review]
Any change should include info about upgrade and migration.

Also since the operator API is likely to need major changes, to remove unused and make consistent with helm - we should evaluate carefully what we want to keep.

File: operator/v1alpha1/operator.proto (hunk 1 above)
Comment (shamsher31) [reply operator/v1alpha1/operator.proto:176]: `BaseComponentSpec` only supported `overlay` and not the complete resource spec. Not sure if backward compatibility matters here?

**Comment by shamsher31** [comment]
Please see the issue https://github.com/istio/istio/issues/27240 and let me know if this PR needs any updates. Thanks.

**Review by richardwxn — CHANGES_REQUESTED** [review]
This change doesn't seem to solve the problem, it would only allow overlays field. I am not sure what went wrong but the original API definition looks correct and aligns with other component, so probably it is not an API issue but somewhere in the operator code.

**Comment by shamsher31** [comment]
Closing as this is not a valid solution.

**Pull request closed without merging**